### PR TITLE
Fix for bug #948

### PR DESCRIPTION
**Pull request by Rakesh-Dakshinamoorthy**
## Description
The slash fails to collect the class based test on rerun or resume. when saving test for resume it saves only the function name. While collecting test it compares with the address in file.
```
    def _iter_test_resume(self, resume_state):
        for test in self._iter_path(resume_state.file_name):
            if resume_state.function_name == test.__slash__.address_in_file:
                if resume_state.variation:
                    if not resume_state.variation == test.get_variation().id:
                        continue
                yield test
```

## Solution
Instead of function name saved the address in file property in ResumeState(function_name = address_in_file)

## Motivation and Context
#948 

## How Has This Been Tested?
```
slash resume random-previous session-id
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have modified tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Rakesh-Dakshinamoorthy** [comment]
The unit tests was comparing the function name but in `_iter_test_resume` actually address_in_file is compared in collecting the tests. Hence modified the tests too since they are not proper

**Comment by ayalash** [comment]
@Rakesh-Dakshinamoorthy Thanks for contributing!

File: slash/resuming.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def save_resume_state(session_result, collected_tests):
 
     for test_metadata in tests_with_no_results:
         test_to_resume = ResumeState(session_id=session_result.session.id, file_name=test_metadata.file_path,\
-                                     function_name=test_metadata.function_name, status=ResumeTestStatus.PLANNED)
+                                     function_name=test_metadata.address_in_file, status=ResumeTestStatus.PLANNED)
```
Comment (ayalash) [inline slash/resuming.py]: Please rename `ResumeState` object's member from `function_name` to `address_in_file`

Comment (ayalash) [reply slash/resuming.py]: In addition, please change the DB version (due to the above rename) by:
```
_DB_NAME = 'resume_state_v2.db'
```
(It is currently `v1`

Comment (Rakesh-Dakshinamoorthy) [reply slash/resuming.py]: sure changed as per @ayalash comments

**Comment by Rakesh-Dakshinamoorthy** [comment]
@ayalash Fixed as per comments please review.

**Comment by vmalloc** [comment]
Thank you for your contribution!

**Comment by Rakesh-Dakshinamoorthy** [comment]
> Thank you for your contribution!

You are welcome.

Also may I know when new release (1.8.1) for slash will be created?
Is there any possibility that a new release (1.8.1) will be created any time soon?

**Comment by vmalloc** [comment]
As it seems at the moment, the next release is going to be 1.9.0 which will contain this fix. If this is very critical to you we can consider cutting 1.8.1 with only this fix, but we generally prefer to avoid it.

@ayalash what do you think?

**Comment by ayalash** [comment]
@vmalloc Yes, I think the best thing will be to release 1.9.0 soon.
@Rakesh-Dakshinamoorthy we are currently running with develop branch on our environment to ensure the new bug fixes and features don't break anything.  I believe we'll be ready to release the 1.9.0 during next week.

**Comment by Rakesh-Dakshinamoorthy** [comment]
> @vmalloc Yes, I think the best thing will be to release 1.9.0 soon.
> @Rakesh-Dakshinamoorthy we are currently running with develop branch on our environment to ensure the new bug fixes and features don't break anything. I believe we'll be ready to release the 1.9.0 during next week.

Thanks @ayalash  and @vmalloc

**Comment by Rakesh-Dakshinamoorthy** [comment]
Hi @ayalash and @vmalloc ,
Could you please let me know when the 1.9.0 will be released?
Is there any possibility it will me released anytime soon.

**Comment by ayalash** [comment]
Hi @Rakesh-Dakshinamoorthy, Our testing took more then expected, but it looks good. We'll release 1.9.0 in the beginning of next week.

**Comment by ayalash** [comment]
@Rakesh-Dakshinamoorthy 1.9.0 has just been released

**Pull request merged**